src/main/java/com/google/api/gax/grpc/AbstractPage.java

```diff
@@ -29,76 +29,81 @@
  */
 package com.google.api.gax.grpc;
 
+import com.google.api.gax.core.Page;
 import com.google.common.base.Preconditions;
 import com.google.common.collect.AbstractIterator;
 import com.google.common.collect.Iterables;
 import java.util.Iterator;
 
-public class PageContext<RequestT, ResponseT, ResourceT> {
+public abstract class AbstractPage<RequestT, ResponseT, ResourceT> implements Page<ResourceT> {
+
+  protected interface PageFactory<RequestT, ResponseT, ResourceT, PageT> {
+    PageT createPage(
+        UnaryCallable<RequestT, ResponseT> callable,
+        PagedListDescriptor<RequestT, ResponseT, ResourceT> pageDescriptor,
+        RequestT request,
+        CallContext context,
+        ResponseT response);
+  }
```

Comment:
I don't really know how this would work, perhaps we can discuss? I have added a PageContext, which I think helps out with code cleanliness.